src/composables/transactionItemList.ts

```diff
@@ -0,0 +1,128 @@
+import { computed } from '@vue/composition-api';
+import {
+  IActiveMultisigTx,
+  IDefaultComposableOptions,
+  ITokenList,
+  ITransaction,
+  ITx,
+} from '../types';
+import {
+  AETERNITY_CONTRACT_ID,
+  defaultTransactionSortingCallback,
+  getInnerTransaction,
+  TX_FUNCTIONS,
+} from '../popup/utils';
+import { useTransactionTx } from './transactionTx';
+import { usePendingMultisigTransaction } from './pendingMultisigTransaction';
+import { useTransactionAndTokenFilter } from './transactionAndTokenFilter';
+import { i18n } from '../store/plugins/languages';
+
+interface TransactionItemList extends IDefaultComposableOptions {
+  isMultisig?: boolean,
+  token: string,
+}
+
+type ITransactionElement = ITransaction | IActiveMultisigTx
+
+export const useTransactionList = ({
+  store,
+  isMultisig,
+  token,
+}: TransactionItemList) => {
+  const availableTokens = computed<ITokenList>(() => store.state.fungibleTokens.availableTokens);
+  const transactions = computed(() => store.state.transactions);
+  const getTxSymbol = computed(() => store.getters.getTxSymbol);
+  const getAccountPendingTransactions = computed<IActiveMultisigTx[]>(
+    () => store.getters.getAccountPendingTransactions,
+  );
+
+  const canLoadMore = computed(() => !!transactions.value.nextPageUrl);
+
+  const {
+    searchPhrase,
+    displayMode,
+    FILTER_MODE,
+  } = useTransactionAndTokenFilter();
+
+  const { pendingMultisigTransaction } = usePendingMultisigTransaction({ store });
+
+  function isFungibleTokenTx(tx: ITx) {
+    return Object.keys(availableTokens.value).includes(tx.contractId);
+  }
+
+  function filterProperTransactions(tr: ITransactionElement) {
+    const innerTx = getInnerTransaction(tr.tx!);
+    return !token || (
+      token !== AETERNITY_CONTRACT_ID
+        ? innerTx?.contractId === token
+        : (!innerTx.contractId || !isFungibleTokenTx(innerTx))
+    );
+  }
+
+  function filterTransactionByDisplayMode(tr: ITransactionElement) {
+    const { direction, isDex } = useTransactionTx({
```

Comment:
I think it's bad thing to use another composable in a loop (this function is a callback for `filter` so it will be fired multiple times.

Reply:
@CedrikNikita Have we better way to checking direction and isDex ?

Reply:
I think we can extract the logic into separate functions.